File: buggy3.go
package main

import "fmt"

func divide(a int, b int) int {
	return a / b
}

func findUser(users map[string]int, name string) int {
	return users[name]
}

func main() {
	users := map[string]int{
		"Jayant": 20,
		"Alex":   21,
	}

	fmt.Println(divide(10, 0))
	fmt.Println(findUser(users, "John"))
}

File: buggy3.rs
fn divide(a: i32, b: i32) -> i32 {
    a / b
}

fn find_user(users: Vec<(&str, i32)>, name: &str) -> i32 {
    for (user, age) in users {
        if user == name {
            return age;
        }
    }

    panic!("User not found");
}

fn main() {
    let users = vec![
        ("Jayant", 20),
        ("Alex", 21),
    ];

    println!("{}", divide(10, 0));
    println!("{}", find_user(users, "John DOE"));
}